src/Comments/CommentList/index.tsx

```diff
@@ -15,3 +20,82 @@ export function CommentList({ comments }: CommentListProps) {
         </div>
     );
 }
+
+export function CommentListWithLinks({ comments }: CommentListProps) {
+    return (
+        <div className={s.container}>
+            {comments.map(comment => {
+                return <CommentWithLink key={comment.id} comment={comment} />;
+            })}
+        </div>
+    );
+}
+
+function CommentWithLink({ comment }: { comment: Comment }) {
+    const getDocument = useDocument(comment.ref);
+    const linkedDocument = getDocument();
+
+    return (
+        <div>
+            <div className={s.linkToDocument}>
+                <LinkToDocument document={linkedDocument} documentRef={comment.ref} creator={comment.creator} />
```

Comment:
I would pass `getDocument` directly here. And then call it inside `LinkToDocument` and wrap it in a Suspense here, so we don't slow down the loading of the list. Ideally it would have a fallback that will most likely have the correct height so we don't have the list jump around.

Reply:
I am wrapping the whole thing in a suspense now, because if the referenced `Term | Translation | TranslationExample `does not exist, I also do not want to show the comment at all. That makes the loading replacement thing a little hard – comments can vary a lot in height. Maybe the whole list should be suspended instead then?

Or, should comments on deleted entities be prevented at another place, e.g. delete comments when the entity is deleted?